### PR TITLE
Fix non aggressive npcs

### PR DESCRIPTION
**Pull request by SamuelllG**
pulled sanity_advanced (merge it first, and you'll have an easy time reviewing)

made non aggressive enemies run away if being hit

### CODE REVIEW

File: Assets/Scripts/Entity/Enemy/EnemyController.cs [hunk 1]
```diff
@@ -158,7 +166,15 @@ private void MoveToNextPoint()
...
 
-        public Area WanderArea => wanderArea;
+        /// <summary>
+        /// Run away. (Move to next point with running speed.)
+        /// </summary>
+        public void RunAway()
+        {
+            _runningAway = true;
+            NavMeshAgent.speed = runningSpeed;
+            MoveToNextPoint();
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Enemy/EnemyController.cs:176]: Probably just a random point in any direction? Or away from the attacker?

File: Assets/Scripts/Entity/Player/Sanity/SanityController.cs [hunk 2]
```diff
@@ -16,64 +14,260 @@ public enum StatType { Health, Saturation, Hydration };
...
 
         private void Awake()
         {
             _playerState = GetComponent<PlayerState>();
+            _attackLogic = GetComponent<AttackLogic>();
         }
         
         private void OnEnable()
         {
             PlayerState.OnPlayerHealthUpdate += (int health) => sanityMath.InfluenceSanityByStat(StatType.Health, health);
```
Comment (DavidPgl) [inline Assets/Scripts/Entity/Player/Sanity/SanityController.cs]: Lambda functions cannot be unsubscribed!

**Review by DavidPgl — DISMISSED** [review]

File: Assets/Scripts/Entity/Enemy/EnemyController.cs (hunk 1 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/Enemy/EnemyController.cs:176]: It's just another random point

File: Assets/Scripts/Entity/Player/Sanity/SanityController.cs (hunk 2 above)
Comment (SamuelllG) [reply Assets/Scripts/Entity/Player/Sanity/SanityController.cs]: Ahh ya I saw that as well, forgot to fix it

**Review by DavidPgl — APPROVED** [review]

**Review by thuemmels — APPROVED** [review]

**Review by Kaes3kuch3n — APPROVED** [review]

**Pull request merged**